### PR TITLE
[DMS-600] StudentSchoolAssociation Authorization: E2E test scenarios

### PR DESCRIPTION
**Pull request by axelmarquezh**

### CODE REVIEW

File: src/dms/tests/EdFi.DataManagementService.Tests.E2E/Features/Authorization/RelationshipsWithEdOrgsAndStudents.feature [hunk 1]
```diff
@@ -341,3 +1387,265 @@ Scenario: 06 Ensure client can not delete or get a PostSecondaryEvent with out s
                       ]
                     }
                 """
+
+Rule: Edge cases are properly authorized
```
Comment (axelmarquezh) [inline src/dms/tests/EdFi.DataManagementService.Tests.E2E/Features/Authorization/RelationshipsWithEdOrgsAndStudents.feature:1391]: These are the scenarios requested in the ticket's description.
I'd welcome a better Rule name .

File: src/dms/tests/EdFi.DataManagementService.Tests.E2E/StepDefinitions/StepDefinitions.cs [hunk 2]
```diff
@@ -31,6 +31,7 @@ ScenarioContext _scenarioContext
         private string _etag = string.Empty;
         private string _dependentId = string.Empty;
         private string _referencedResourceId = string.Empty;
+        private ScenarioVariables _scenarioVariables = new();
```
Comment (simpat-adam) [inline src/dms/tests/EdFi.DataManagementService.Tests.E2E/StepDefinitions/StepDefinitions.cs:34]: Very handy.

**Review by simpat-adam — APPROVED** [review]
